script.js

```diff
@@ -64,7 +72,21 @@ function createProductListing(category) {
             image = document.createElement('img')
             image.src = product.image
             div.appendChild(image)
-        }    
+            
+            let p = document.createElement('p')
+            productsListing.appendChild(p)
+            p.innerHTML= product.description
+            div.appendChild(p)
+
+            let button = document.createElement('button')
+            productsListing.appendChild(button)
+             button.innerHTML = 'Cart'
+             button.onclick = function(){
+                
+            
+            div.appendChild(button)
+             }  
```

Comment:
This button's onclick handler is a function which, when invoked, will append the button to the parent div.  I'm not exactly sure what this is meant to do but probably this is not accurate.  

If you put a console.log() into this onclick function you can see when it is invoked and then replace its contents with the behavior that you want.  You mentioned a "popup cart" so maybe this button's purpose is, when clicked, to create a floating div with the user's cart contents?

You likely want the div.appendChild() on line 87 to move to line 83 or line 89, ie outside the onclick handler.